### PR TITLE
[READY]  test preview group and question

### PR DESCRIPTION
**Pull request by Shnoulle**
Fixed issue #14995: Must add test for Group/Question relevance

### CODE REVIEW

**Comment by Shnoulle** [comment]
@olleharstedt : did you see any reason why this pull request can send
````
1) ls\tests\RemoteControlTest::testAddResponse
Failed asserting that Array &0 (
    'status' => 'Invalid user name or password'
) is not equal to Array &0 (
    'status' => 'Invalid user name or password'
).
````

Else : 
1. Did you think it's better to create 4 test for each url ? https://github.com/LimeSurvey/LimeSurvey/pull/1310/files#diff-f8ff6b4974f61d5543dc9ac219bad2d5R38
2. Or to create 4 function to call inside the test ?
3. Or another way ?

Do you know the way to save a picture on all error ? Without using multiple Catch ?

File: tests/surveys/PreviewGroupAndQuestionTest.php [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        $urlMan->setBaseUrl('http://' . self::$domain . '/index.php');
+        $url = $urlMan->createUrl(
+            'survey/index',
+            [
+                'sid' => self::$surveyId,
+                'action' => 'previewgroup',
+                'gid' => $questions['G2Q01']['gid'],
+            ]
+        );
+        try {
```
Comment (Shnoulle) [inline tests/surveys/PreviewGroupAndQuestionTest.php]: 1st url call

File: tests/surveys/PreviewGroupAndQuestionTest.php [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        $url = $urlMan->createUrl(
+            'survey/index',
+            [
+                'sid' => self::$surveyId,
+                'action' => 'previewgroup',
+                'gid' => $questions['G2Q01']['gid'],
+                'Q02' => 'Y',
+            ]
+        );
+        try {
```
Comment (Shnoulle) [inline tests/surveys/PreviewGroupAndQuestionTest.php]: 2nd url call

File: tests/surveys/PreviewGroupAndQuestionTest.php [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        $url = $urlMan->createUrl(
+            'survey/index',
+            [
+                'sid' => self::$surveyId,
+                'action' => 'previewquestion',
+                'gid' => $questions['G3Q02']['gid'],
+                'qid' => $questions['G3Q02']['qid'],
+            ]
+        );
+        try {
```
Comment (Shnoulle) [inline tests/surveys/PreviewGroupAndQuestionTest.php]: 3rd url call

File: tests/surveys/PreviewGroupAndQuestionTest.php [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+            'survey/index',
+            [
+                'sid' => self::$surveyId,
+                'action' => 'previewquestion',
+                'gid' => $questions['G3Q02']['gid'],
+                'qid' => $questions['G3Q02']['qid'],
+                'Q03' => "Y",
+            ]
+        );
+        try {
```
Comment (Shnoulle) [inline tests/surveys/PreviewGroupAndQuestionTest.php]: 4th url call

**Comment by olleharstedt** [comment]
> Do you know the way to save a picture on all error ? Without using multiple Catch ?

Not necessary. The `assertFalse` will halt the test anyway, so only one pic will be saved.

**Comment by Shnoulle** [comment]
> Not necessary. The assertFalse will halt the test anyway, so only one pic will be saved.

Yes, but no pic saved if test is only broken (for example a element is not displayed). 
Yesterday : i try to found the good way to check a radio button … just have a screenshot to see the radio button is not checked can help me.

I have to manually set the screenshot after the error.

**Comment by olleharstedt** [comment]
Use multiple methods then.

**Comment by Shnoulle** [comment]
> Use multiple methods then.

? 4 different test for now.

Just don't understand why logout broke travis before the test …

**Comment by olleharstedt** [comment]
Now it's the proper error again. The hunts continues - tomorrow!

**Comment by Shnoulle** [comment]
Then need to be done before RC ?

But think RC test muts be fixed to allow anay test before itself :)

**Comment by Shnoulle** [comment]
@olleharstedt got it : plugin loaded only if not already installed. Loaded all times : fixed.

**Comment by olleharstedt** [comment]
https://www.youtube.com/watch?v=EqWRaAF6_WY

**Pull request merged**